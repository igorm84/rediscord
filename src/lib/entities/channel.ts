import { Component, ReactComponentElement } from "react";
import { AiFillSound } from "react-icons/ai";
import { BsHash } from "react-icons/bs";

// here keys represent the type of channel
export const channelIcons = {
  "text": BsHash,
  "voice": AiFillSound,
}
export interface Channel {
  id: string;
  title: string;
  slug: string;
  type: keyof typeof channelIcons;
}
export interface ChannelGroup {
  id: string;
  title: string;
  channels: Channel[];
}
