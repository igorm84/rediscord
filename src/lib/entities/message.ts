export interface Message {
  id: string;
  replyMessageId?: string;
  authorNickname: string;
  authorAvatar: string;
  message: string;
  timestamp: number;
}
