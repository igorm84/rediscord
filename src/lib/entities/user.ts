import { Database } from "../db/database.types";
import { Activity } from "./activity";

export enum UserStatuses {
  online = "online",
  idle = "idle",
  dnd = "dnd",
  offline = "offline",
  mobile = "mobile",
}
export interface VoiceStatus {
  mute?: boolean;
  deaf?: boolean;
  serverMuted?: boolean;
}

export type User = Omit<
  Database["public"]["Tables"]["profiles"]["Row"],
  "status"
> & {
  activities?: Activity[];
  status: UserStatuses;
};
