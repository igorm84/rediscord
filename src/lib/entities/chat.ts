import { UserStatuses } from "./user";
import { Activity } from "./activity";

export type Chat = {
  id: string;
  name: string;
  status: UserStatuses;
  activity?: Activity | null;
  avatar?: string | null;
};
