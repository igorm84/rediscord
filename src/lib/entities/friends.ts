import { Database } from "../db/database.types";
import { User } from "./user";

export type Friend = Partial<
  Database["public"]["Tables"]["friendships"]["Row"]
> & {
  user?: User;
};
