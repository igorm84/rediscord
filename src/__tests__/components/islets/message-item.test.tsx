import { Message } from "@/lib/entities/message";
import { render, screen } from "@testing-library/react";
import { composeStory } from "@storybook/react";
import Meta, {
    Default,
} from "@/components/islets/message-item/message-item.stories";
import dayjs from "dayjs";

const DefaultMessage = composeStory(Default, Meta);
const props: Message = {
    id: "1",
    authorAvatar: "https://i.pravatar.cc/150?img=1",
    authorNickname: "John Doe",
    message:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed euismod, velit vel bibendum bibendum, nisl elit bibendum sapien, vel bibendum sapien elit vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien. Sed vel sapien vel sapien.",
    timestamp: Date.now(),
};
test("renders message item", () => {
    render(<DefaultMessage {...props} />);
    expect(screen.getByRole("message-item")).toBeInTheDocument();
});

test("renders message item with correct author nickname", () => {
    render(<DefaultMessage {...props} />);
    expect(screen.getByText(props.authorNickname)).toBeInTheDocument();
});

test("renders message item with correct message", () => {
    render(<DefaultMessage {...props} />);
    expect(screen.getByText(props.message)).toBeInTheDocument();
});

test("renders message item with correct timestamp", () => {
    render(<DefaultMessage {...props} />);
    const date = dayjs(props.timestamp);
    const hDiff = dayjs(Date.now()).diff(props.timestamp) / 1000 / 60 / 60;
    const expected =
        hDiff >= 23 ? date.format("MMMM D, YYYY h:mm A") : date.format("HH:mm A");
    expect(screen.getByText(expected)).toBeInTheDocument();
});
