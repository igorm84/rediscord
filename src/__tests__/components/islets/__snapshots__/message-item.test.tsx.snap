// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders message item 1`] = `
<div
  class="grid min-h-[35px] cursor-pointer grid-cols-[35px_1fr] 
    items-start gap-2 bg-transparent px-4 py-3 hover:bg-midground"
>
  <div
    class="relative flex h-8 w-8 items-center justify-center rounded-full text-white h-8 w-8 bg-white/5"
  >
    <img
      alt="avatar"
      class="rounded-full"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="32"
      src="https://i.pravatar.cc/150?img=3"
      style="color: transparent;"
      width="32"
    />
  </div>
  <div
    class="grid grid-rows-[13px_1fr] gap-[5px]"
  >
    <div
      class="flex gap-[6px]"
    >
      <span
        class="grow-0 text-xs font-semibold text-white"
      >
        John Doe
      </span>
      <span
        class="grow-0 text-xs font-medium text-zinc-500"
      >
        21:57 PM
      </span>
    </div>
    <p
      class="text-xs font-medium text-zinc-300"
    >
      Hello, world!
    </p>
  </div>
</div>
`;
